### PR TITLE
fixing GroupVersionForDiscovery struct comment in staging/src/k8s.io/…

### PR DESCRIPTION
**Pull request by huangjiuyuan**
**What this PR does / why we need it**:
This PR fix the comment of GroupVersionForDiscovery struct in `staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go`.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

`NONE`

### CODE REVIEW

File: staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go [hunk 1]
```diff
@@ -758,7 +758,7 @@ type ServerAddressByClientCIDR struct {
 	ServerAddress string `json:"serverAddress" protobuf:"bytes,2,opt,name=serverAddress"`
 }
 
-// GroupVersion contains the "group/version" and "version" string of a version.
+// GroupVersionForDiscovery contains the "group/version" and "version" string of a version.
```
Comment (dixudx) [inline staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go]: Please update the auto-generated files as well.

Comment (huangjiuyuan) [reply staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go]: @dixudx Thanks for your feedback. Could you please tell me how to update the auto-generated files? I had run `hack/update-generated-docs.sh`.

Comment (dixudx) [reply staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go]: @huangjiuyuan run `make generated_files`

Comment (dixudx) [reply staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go]: Related proto and json files has to be updated. Please run `./hack/update-all.sh` instead.

Comment (huangjiuyuan) [reply staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go]: @dixudx I see. A `generated.proto` file had been updated. Good to go?

**Comment by huangjiuyuan** [comment]
@smarterclayton Could you please approve this? Thank you!

**Comment by huangjiuyuan** [comment]
@smarterclayton Could you please approve this one? It seems this PR need a `/approve no-issue` to get approved. Thank you!

**Pull request closed without merging**